raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -695,6 +682,111 @@ public void testObserverFindQuorumAfterElectionTimeout() throws IOException {
         assertSentFindQuorumRequest();
     }
 
+    @Test
+    public void testInvalidFetchRequest() throws Exception {
```

Comment:
Nice added coverage! Do we have coverage for the LogTruncationException logic as well? I ran the coverage tool but didn't it gets triggered.

Reply:
This is a gap I am hoping we can address when we add the `truncate` API to `ReplicatedStateMachine`.

Reply:
Sg, will get a JIRA.